### ISSUE TITLE
Need to handle or reject combined logical operator and expressions

### ISSUE THREAD

**Issue by mqudsi** [fish-shell/fish-shell#9403]
What should be the result of the following:

```fish
echo foo && and echo bar;
```

It currently prints

```
foo
```

The options I see would be to print 

```
foo
bar
```

or to error out at the parser level (before anything is run). I would lean towards the latter.

**Comment by faho** [comment]
>to error out at the parser level (before anything is run). I would lean towards the latter.

Yeah that seems better. I can't see how anything else is intuitive.

A possible error would be the same as for `echo && && echo`:

> fish: Expected a command, but found '&&'

which is simple and parsery, we might be able to find better wording.
